WalletWasabi/JsonConverters/PubKeyJsonConverter.cs

```diff
@@ -3,24 +3,18 @@
 
 namespace WalletWasabi.JsonConverters;
 
-public class PubKeyJsonConverter : JsonConverter
+public class PubKeyJsonConverter : JsonConverter<PubKey>
 {
 	/// <inheritdoc />
-	public override bool CanConvert(Type objectType)
+	public override PubKey? ReadJson(JsonReader reader, Type objectType, PubKey? existingValue, bool hasExistingValue, JsonSerializer serializer)
 	{
-		return objectType == typeof(PubKey);
+		return new PubKey(((string?)reader.Value)?.Trim());
```

Comment:
This reports a warning for me.

Reply:
Not perfect but this does not report warnings:

```cs
	/// <inheritdoc />
	public override PubKey? ReadJson(JsonReader reader, Type objectType, PubKey? existingValue, bool hasExistingValue, JsonSerializer serializer)
	{
		string? hex = ((string?)reader.Value)?.Trim();

		if (hex is null)
		{
			throw new ArgumentNullException(nameof(hex));
		}

		return new PubKey(hex);
	}
```

Reply:
Done.